=================================================  

Program ini dibuat untuk melakukan web scrapping mencari produk seblak dan melakukan analisis data. Melakukan analisis data dengan menggunakan rumus-rumus statistika yang nantinya digunakan untuk mencari tahu apakah penjualan produk seblak bisa mendapatkan banyak keuntungan agar mendapatkan pendapatan tambahan.


=================================================



## Web Scrapping

In [7]:
# import libraries
import pandas as pd
from selenium import webdriver
from bs4 import BeautifulSoup
driver = webdriver.Chrome()
import time

In [20]:
# Menjalankan scrapping
dataset = []
# Melakukan looping
for i in range(1, 11):
    url = f'https://www.tokopedia.com/search?navsource=&page={i}&q=seblak&srp_component_id=02.01.00.00&srp_page_id=&srp_page_title=&st='
    driver.get(url)
    time.sleep(2)
    html = driver.page_source
    soup = BeautifulSoup(html, "html.parser")
    datalist = soup.findAll("div", {"class":"css-1asz3by"})

    # Looping
    for data in datalist:
        try:
            dict = {
                "nama_produk": data.find("div", {"class": "prd_link-product-name"}).getText(),
                "harga_produk": data.find("div", {"class": "prd_link-product-price"}).getText(),
                "penjual": data.find("span", {"class": "prd_link-shop-name"}).getText(),
                "kota_toko": data.find("span", {"class": "prd_link-shop-loc"}).getText(),
                "banyaknya_terjual": data.find("span", {"class": "prd_label-integrity"}).getText(),
                "rating_produk": data.find("span", {"class": "prd_rating-average-text"}).getText()
            }
            dataset.append(dict)
        except:
            pass
    print(len(dataset))
print(len(dataset))

35
70
103
138
173
207
242
276
311
346
346


In [21]:
# Membuat sebuah variabel df dan mengubah dataset ke pandas dataframe
df = pd.DataFrame(dataset)
df

nama_produk harga_produk  \
0    Gelifood Seblak Instan Kerupuk Mawar Bumbu Ken...     Rp13.000   
1    Seblak Instan Ceu Nthien Khas Bandung Rasana N...     Rp17.000   
2              Seblak Rafael, Seblak Coet Instan Halal     Rp25.000   
3                    Kylafood Seblak Cup Instan, Halal     Rp11.500   
4                    KERUPUK SEBLAK MENTAH ANEKA WARNA     Rp11.500   
..                                                 ...          ...   
341                      Seblak Instan Pedas Home Made      Rp3.500   
342  Gelifood Seblak Instan Kerupuk Mawar Bumbu Ken...     Rp13.000   
343  keripik beling/kaca 1/2 kg. bumbu seblak daun ...     Rp25.999   
344             Seblak pedas kerupuk pedas kiloan 1 kg     Rp35.000   
345           Seblak Instan Komplit Pedas Gurih Nikmat      Rp5.999   

                      penjual          kota_toko banyaknya_terjual  \
0                    Lidigeli         Kab. Garut      250+ terjual   
1    Central Seblak Nusantara  Tangerang Selatan      2rb+ terjual   
2           Brother Meat Shop              Depok      100+ terjual   
3           Brother Meat Shop              Depok       60+ terjual   
4              jajanangarut19    Jakarta Selatan      250+ terjual   
..                        ...                ...               ...   
341                the Dhecip  Tangerang Selatan      2rb+ terjual   
342                  Lidigeli         Kab. Garut      250+ terjual   
343                   aghniya  Tangerang Selatan      250+ terjual   
344             RIKI STORE 97         Kab. Bogor      100+ terjual   
345            Baso Aci Ayang         Kab. Garut      2rb+ terjual   

    rating_produk  
0             4.8  
1             4.9  
2             4.9  
3             5.0  
4             4.9  
..            ...  
341           4.8  
342           4.8  
343           4.9  
344           4.8  
345           4.9  

[346 rows x 6 columns]

Berhasil melakukan scraping dan mendapatkan hasil yang sudah diubah ke pandas dataframe.

In [22]:
# Export ke csv
df.to_csv("SeblakDataRaw.csv", index=False)

Melakukan export dataframe ke csv dengan nama "SeblakDataRaw.csv"

## Data Preparation

### Data Knowledge

In [4]:
# Membaca data csv
dataRaw = pd.read_csv("SeblakDataRaw.csv")
print(dataRaw.info())
dataRaw.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 346 entries, 0 to 345
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   nama_produk        346 non-null    object 
 1   harga_produk       346 non-null    object 
 2   penjual            346 non-null    object 
 3   kota_toko          346 non-null    object 
 4   banyaknya_terjual  346 non-null    object 
 5   rating_produk      346 non-null    float64
dtypes: float64(1), object(5)
memory usage: 16.3+ KB
None


nama_produk harga_produk  \
0  Gelifood Seblak Instan Kerupuk Mawar Bumbu Ken...     Rp13.000   
1  Seblak Instan Ceu Nthien Khas Bandung Rasana N...     Rp17.000   
2            Seblak Rafael, Seblak Coet Instan Halal     Rp25.000   
3                  Kylafood Seblak Cup Instan, Halal     Rp11.500   
4                  KERUPUK SEBLAK MENTAH ANEKA WARNA     Rp11.500   

                    penjual          kota_toko banyaknya_terjual  \
0                  Lidigeli         Kab. Garut      250+ terjual   
1  Central Seblak Nusantara  Tangerang Selatan      2rb+ terjual   
2         Brother Meat Shop              Depok      100+ terjual   
3         Brother Meat Shop              Depok       60+ terjual   
4            jajanangarut19    Jakarta Selatan      250+ terjual   

   rating_produk  
0            4.8  
1            4.9  
2            4.9  
3            5.0  
4            4.9

Terdapat 6 kolom dan 346 produk

In [34]:
# Mencari unik value pada kolom kota_toko
print(dataRaw['kota_toko'].unique())
len(dataRaw['kota_toko'].unique())

['Kab. Garut' 'Tangerang Selatan' 'Depok' 'Jakarta Selatan'
 'Jakarta Timur' 'Kab. Bogor' 'Jakarta Barat' 'Bekasi' 'Surakarta'
 'Kab. Bekasi' 'Kab.Ciamis' 'Jakarta Pusat' 'Cimahi' 'Palembang'
 'Kab. Tangerang' 'Surabaya' 'Bandung' 'Banjarbaru' 'Tasikmalaya'
 'Tangerang' 'Kab. Bandung' 'Kab. Sumedang' 'Semarang' 'Kab. Gresik'
 'Jakarta Utara']


25

Diketahui terdapat 25 kota pada data.

### Data Cleaning

In [44]:
# Melakukan pengecekan null values
dataRaw.isnull().any()

nama_produk          False
harga_produk         False
penjual              False
kota_toko            False
banyaknya_terjual    False
rating_produk        False
dtype: bool

Tidak terdapat null value pada data

In [10]:
# Melakukan import library
import pandas as pd

In [2]:
# Melakukan pembacaan data csv dan menghilangkan symbol
dataRaw = pd.read_csv("SeblakDataRaw.csv")
dataRaw["harga_produk"] = dataRaw["harga_produk"].str.replace("Rp", "").str.replace(".", "").astype(int)
dataRaw["banyaknya_terjual"] = dataRaw["banyaknya_terjual"].str.replace("rb", "000").str.replace("+", "").str.replace(" terjual", "").astype(int)
dataRaw.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 346 entries, 0 to 345
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   nama_produk        346 non-null    object 
 1   harga_produk       346 non-null    int32  
 2   penjual            346 non-null    object 
 3   kota_toko          346 non-null    object 
 4   banyaknya_terjual  346 non-null    int32  
 5   rating_produk      346 non-null    float64
dtypes: float64(1), int32(2), object(3)
memory usage: 13.6+ KB


Melakukan cleaning dengan cara menghapus simbol "Rp", ".", "rb", "000", "+", "terjual" dan merubah data typenya ke integer.

In [7]:
# Mencari nilai unik dari kolom kota_toko
print(dataRaw['kota_toko'].unique())

['Kab. Garut' 'Tangerang Selatan' 'Depok' 'Jakarta Selatan'
 'Jakarta Timur' 'Kab. Bogor' 'Jakarta Barat' 'Bekasi' 'Surakarta'
 'Kab. Bekasi' 'Kab.Ciamis' 'Jakarta Pusat' 'Cimahi' 'Palembang'
 'Kab. Tangerang' 'Surabaya' 'Bandung' 'Banjarbaru' 'Tasikmalaya'
 'Tangerang' 'Kab. Bandung' 'Kab. Sumedang' 'Semarang' 'Kab. Gresik'
 'Jakarta Utara']


Diketahui bahwa terdapat 25 kota yang berbeda dari data.

In [6]:
# Membuat sebuah kolom baru untuk kota-kota yang ada di jabodetabek dan luar jabodetabek.
jabodetabek_list = []
for kota in dataRaw['kota_toko']:
    if kota in ['Tangerang Selatan', 'Depok', 'Jakarta Selatan','Jakarta Timur', 
               'Kab. Bogor', 'Jakarta Barat', 'Bekasi', 'Kab. Bekasi', 'Jakarta Pusat', 'Kab. Tangerang', 'Tangerang', 'Jakarta Utara']:
        jabodetabek_list.append('Jabodetabek')
    else:
        jabodetabek_list.append('Luar Jabodetabek')

dataRaw['jabodetabek'] = jabodetabek_list
dataRaw

nama_produk  harga_produk  \
0    Gelifood Seblak Instan Kerupuk Mawar Bumbu Ken...         13000   
1    Seblak Instan Ceu Nthien Khas Bandung Rasana N...         17000   
2              Seblak Rafael, Seblak Coet Instan Halal         25000   
3                    Kylafood Seblak Cup Instan, Halal         11500   
4                    KERUPUK SEBLAK MENTAH ANEKA WARNA         11500   
..                                                 ...           ...   
341                      Seblak Instan Pedas Home Made          3500   
342  Gelifood Seblak Instan Kerupuk Mawar Bumbu Ken...         13000   
343  keripik beling/kaca 1/2 kg. bumbu seblak daun ...         25999   
344             Seblak pedas kerupuk pedas kiloan 1 kg         35000   
345           Seblak Instan Komplit Pedas Gurih Nikmat          5999   

                      penjual          kota_toko  banyaknya_terjual  \
0                    Lidigeli         Kab. Garut                250   
1    Central Seblak Nusantara  Tangerang Selatan               2000   
2           Brother Meat Shop              Depok                100   
3           Brother Meat Shop              Depok                 60   
4              jajanangarut19    Jakarta Selatan                250   
..                        ...                ...                ...   
341                the Dhecip  Tangerang Selatan               2000   
342                  Lidigeli         Kab. Garut                250   
343                   aghniya  Tangerang Selatan                250   
344             RIKI STORE 97         Kab. Bogor                100   
345            Baso Aci Ayang         Kab. Garut               2000   

     rating_produk       jabodetabek  
0              4.8  Luar Jabodetabek  
1              4.9       Jabodetabek  
2              4.9       Jabodetabek  
3              5.0       Jabodetabek  
4              4.9       Jabodetabek  
..             ...               ...  
341            4.8       Jabodetabek  
342            4.8  Luar Jabodetabek  
343            4.9       Jabodetabek  
344            4.8       Jabodetabek  
345            4.9  Luar Jabodetabek  

[346 rows x 7 columns]

Membuat sebuah kolom baru bernama jabodetabek dan mengklasifikasikan produk dengan kota jabodetabek dan luar jabodetabek

In [8]:
# Export ke csv
dataRaw.to_csv("SeblakDataClean.csv", index=False)

Export data ke csv

## Business Understanding/Problem Statement

Untuk bisa mendapatkan pendapatan tambahan, saya mencoba mencari cara salah satunya adalah dengan berjualan. Karena modal yang dibutuhkan untuk berjualan lumayan besar dan saya tidak memiliki modal yang banyak, saya memutuskan untuk melakukan dropship. Salah satu produk yang ingin saya lakukan dropship adalah produk seblak. Karena, produk seblak sedang viral yang mana pastinya akan lebih mudah untuk menjualnya. Untuk mendapatkan data yang lebih jelas dan pasti mengenai animo masyarakat dengan seblak serta memperkirakan keuntungan, pertama-tama saya ingin membuat SMART framework sebelum memulai analisis data.  

Specific = Mendapatkan keuntungan lebih dari menjual produk seblak.  

Measurable = Bisa meningkatkan pendapatan hingga 30% dari menjual seblak.

Achiveable = Dengan menganalisa data yang didapat, saya bisa melakukan kalkulasi terkait seberapa besar profit yang akan saya dapatkan jika menjalankan dropship seblak.

Relevant = Karena seblak yang sedang viral dan memiliki potensi penjualan yang besar pada saat ini, sangat mendukung tujuan saya untuk mendapatkan pendapatan lebih dari menjual seblak.

Time-bound = Satu bulan saya sudah bisa mendapatkan pendapatan tambahan dari dropshipping seblak.

Mengetahui apakah produk seblak yang sedang viral memiliki animo yang besar di masyarakat. Dengan cara, melakukan pengumpulan data hingga melakukan analisa terhadap data yang di dapat, agar bisa memperkirakan seberapa besar profit yang bisa didapat jika melakukan dropship seblak. Dalam jangka waktu satu bulan diperkirakan sudah bisa mendapatkan pendapatan tambahan dari dropship seblak.

## Analysis

In [2]:
# Melakukan import library
import numpy as np
from scipy import stats

In [3]:
# Melakukan pembacaan data dari csv
dataClean = pd.read_csv("SeblakDataClean.csv")
dataClean

nama_produk  harga_produk  \
0    Gelifood Seblak Instan Kerupuk Mawar Bumbu Ken...         13000   
1    Seblak Instan Ceu Nthien Khas Bandung Rasana N...         17000   
2              Seblak Rafael, Seblak Coet Instan Halal         25000   
3                    Kylafood Seblak Cup Instan, Halal         11500   
4                    KERUPUK SEBLAK MENTAH ANEKA WARNA         11500   
..                                                 ...           ...   
341                      Seblak Instan Pedas Home Made          3500   
342  Gelifood Seblak Instan Kerupuk Mawar Bumbu Ken...         13000   
343  keripik beling/kaca 1/2 kg. bumbu seblak daun ...         25999   
344             Seblak pedas kerupuk pedas kiloan 1 kg         35000   
345           Seblak Instan Komplit Pedas Gurih Nikmat          5999   

                      penjual          kota_toko  banyaknya_terjual  \
0                    Lidigeli         Kab. Garut                250   
1    Central Seblak Nusantara  Tangerang Selatan               2000   
2           Brother Meat Shop              Depok                100   
3           Brother Meat Shop              Depok                 60   
4              jajanangarut19    Jakarta Selatan                250   
..                        ...                ...                ...   
341                the Dhecip  Tangerang Selatan               2000   
342                  Lidigeli         Kab. Garut                250   
343                   aghniya  Tangerang Selatan                250   
344             RIKI STORE 97         Kab. Bogor                100   
345            Baso Aci Ayang         Kab. Garut               2000   

     rating_produk       jabodetabek  
0              4.8  Luar Jabodetabek  
1              4.9       Jabodetabek  
2              4.9       Jabodetabek  
3              5.0       Jabodetabek  
4              4.9       Jabodetabek  
..             ...               ...  
341            4.8       Jabodetabek  
342            4.8  Luar Jabodetabek  
343            4.9       Jabodetabek  
344            4.8       Jabodetabek  
345            4.9  Luar Jabodetabek  

[346 rows x 7 columns]

### Analisis Harga Produk

In [10]:

# Menghitung nilai rata-rata dari kolom harga_produk dan memasukkannya ke dalam variabel mean
mean = dataClean.harga_produk.mean()

# Menghitung nilai median dari kolom harga_produk dan memasukkannya ke dalam variabel median
median = dataClean.harga_produk.median()

# Menghitung nilai standar deviasi dari kolom harga_produk dan memasukkannya ke dalam variabel modus
std_dev = dataClean.harga_produk.std()

print(f'Rata-rata harga produk adalah {mean}')
print(f'Median harga produk adalah {median}')
print(f'Standar Deviasi dari harga produk adalah {std_dev}')

# Melakukan pencarian skewness
print(f"Skewness: {dataClean['harga_produk'].skew()}")
# Melakukan pencarian kurtosis
print(f"Kurtosis: {dataClean['harga_produk'].kurtosis()}")
     

Rata-rata harga produk adalah 18060.904624277457
Median harga produk adalah 14999.0
Standar Deviasi dari harga produk adalah 14233.310073617693
Skewness: 2.205659881056513
Kurtosis: 8.986951040870522


Diketahui berdasarkan hasil di atas bahwa kolom harga_produk memiliki nilai rata-rata = 18060.904624277457 dengan nilai median = 14999.0  dan standar deviasi = 14233.310073617693. Terlihat bahwa median lebih rendah dari rata-rata, kemungkinan terdapat nilai tinggi yang mempengaruhi rata-rata. Standar deviasi yang memiliki nilai tinggi menunjukkan variasi besar dalam harga produk. Dari data tersebut dapat diambil kesimpulan bahwa rata-rata harga seblak adalah Rp.18.060. dengan harga tengahnya adalah Rp.14.999.

Dari perhitungan di atas juga dapat dilihat bahwa skewnessnya adalah 2.205659881056513 yang berarti distribusi condong ke kanan atau positif.

Nilai kurtosisnya adalah 8.986951040870522 yang menggambarkan leptokurtic karena memiliki ekor yang lebih berat dibandingkan distribusi normal. Kemungkinan terdapat outlier yang signifikan.

### Analisis Banyaknya Produk Terjual

In [11]:

# Menghitung nilai rata-rata dari kolom banyaknya_terjual dan memasukkannya ke dalam variabel mean
mean = dataClean.banyaknya_terjual.mean()

# Menghitung nilai median dari kolom banyaknya_terjual dan memasukkannya ke dalam variabel median
median = dataClean.banyaknya_terjual.median()

# Menghitung nilai standar deviasi dari kolom banyaknya_terjual dan memasukkannya ke dalam variabel modus
std_dev = dataClean.banyaknya_terjual.std()

print(f'Rata-rata banyaknya produk terjual adalah {mean}')
print(f'Median banyaknya produk terjual adalah {median}')
print(f'Standar Deviasi dari banyaknya produk terjual adalah {std_dev}')

# Melakukan pencarian skewness
print(f"Skewness: {dataClean['banyaknya_terjual'].skew()}")
# Melakukan pencarian kurtosis
print(f"Kurtosis: {dataClean['banyaknya_terjual'].kurtosis()}")
     

Rata-rata banyaknya produk terjual adalah 1083.4942196531792
Median banyaknya produk terjual adalah 100.0
Standar Deviasi dari banyaknya produk terjual adalah 2453.319182273563
Skewness: 3.106623996291119
Kurtosis: 8.50486318583266


Diketahui berdasarkan hasil di atas bahwa kolom banyaknya_terjual memiliki nilai rata-rata = 1083.4942196531792 dengan nilai median = 100.0  dan standar deviasi = 2453.319182273563. Dapat diambil kesimpulan bahwa rata-rata jumlah penjualan seblak adalah 1083.4942196531792 dengan median 100.0 yang berarti terdapat variasi yang besar pada penjualan. Standar deviasinya adalah 2453.319182273563 yang berarti beberapa produk memiliki penjualan yang jauh di atas atau di bawah rata-rata.

Skewnessnya bernilai 3.106623996291119 yang berarti distribusi condong ke arah kanan atau positif.

Nilai kurtosisnya adalah 8.50486318583266 yang menggambarkan leptokurtic sehingga mengindikasikan bahwa terdapat outlier yang signifikan.

### Analisis Rating Produk

In [12]:

# Menghitung nilai rata-rata dari kolom rating_produk dan memasukkannya ke dalam variabel mean
mean = dataClean.rating_produk.mean()

# Menghitung nilai median dari kolom rating_produk dan memasukkannya ke dalam variabel median
median = dataClean.rating_produk.median()

# Menghitung nilai standar deviasi dari kolom rating_produk dan memasukkannya ke dalam variabel modus
std_dev = dataClean.rating_produk.std()

print(f'Rata-rata rating produk adalah {mean}')
print(f'Median rating produk adalah {median}')
print(f'Standar Deviasi dari rating produk adalah {std_dev}')

# Melakukan pencarian skewness
print(f"Skewness: {dataClean['rating_produk'].skew()}")
# Melakukan pencarian kurtosis
print(f"Kurtosis: {dataClean['rating_produk'].kurtosis()}")
     

Rata-rata rating produk adalah 4.889306358381503
Median rating produk adalah 4.9
Standar Deviasi dari rating produk adalah 0.09184791345387558
Skewness: -1.0054009847658465
Kurtosis: 4.502734849069931


Rata-rata rating pada produk seblak adalah 4.889306358381503 yang menunjukan bahwa produk memiliki mayoritas penilaian positif dari pelanggan. Mediannya adalah 4.9 yang berarti ratingnya sangat positif. Standar deviasinya 0.09184791345387558 sangat rendah, yang berarti rating produk tidak terlalu jauh dari nilai rata-rata. 

Skewness bernilai -1.0054009847658465 yang menandakan bahwa distribusinya condong ke arah kiri atau positif.

Nilai kurtosisnya adalah 4.502734849069931 yang menggambarkan leptokurtic. Mungkin terdapat outlier tapi tidak terlalu signifikan.

### Menghitung minimum dan maksimum pendapatan dari menjual seblak

In [5]:
# Menambahkan kolom baru yaitu omset yang berisi perkalian antara kolom harga_produk dan banyaknya_terjual
dataClean = dataClean.assign(omset = dataClean['harga_produk'] * dataClean['banyaknya_terjual'])

# pengecekkan kembali dataframe1
dataClean

nama_produk  harga_produk  \
0    Gelifood Seblak Instan Kerupuk Mawar Bumbu Ken...         13000   
1    Seblak Instan Ceu Nthien Khas Bandung Rasana N...         17000   
2              Seblak Rafael, Seblak Coet Instan Halal         25000   
3                    Kylafood Seblak Cup Instan, Halal         11500   
4                    KERUPUK SEBLAK MENTAH ANEKA WARNA         11500   
..                                                 ...           ...   
341                      Seblak Instan Pedas Home Made          3500   
342  Gelifood Seblak Instan Kerupuk Mawar Bumbu Ken...         13000   
343  keripik beling/kaca 1/2 kg. bumbu seblak daun ...         25999   
344             Seblak pedas kerupuk pedas kiloan 1 kg         35000   
345           Seblak Instan Komplit Pedas Gurih Nikmat          5999   

                      penjual          kota_toko  banyaknya_terjual  \
0                    Lidigeli         Kab. Garut                250   
1    Central Seblak Nusantara  Tangerang Selatan               2000   
2           Brother Meat Shop              Depok                100   
3           Brother Meat Shop              Depok                 60   
4              jajanangarut19    Jakarta Selatan                250   
..                        ...                ...                ...   
341                the Dhecip  Tangerang Selatan               2000   
342                  Lidigeli         Kab. Garut                250   
343                   aghniya  Tangerang Selatan                250   
344             RIKI STORE 97         Kab. Bogor                100   
345            Baso Aci Ayang         Kab. Garut               2000   

     rating_produk       jabodetabek     omset  
0              4.8  Luar Jabodetabek   3250000  
1              4.9       Jabodetabek  34000000  
2              4.9       Jabodetabek   2500000  
3              5.0       Jabodetabek    690000  
4              4.9       Jabodetabek   2875000  
..             ...               ...       ...  
341            4.8       Jabodetabek   7000000  
342            4.8  Luar Jabodetabek   3250000  
343            4.9       Jabodetabek   6499750  
344            4.8       Jabodetabek   3500000  
345            4.9  Luar Jabodetabek  11998000  

[346 rows x 8 columns]

Membuat sebuah kolom baru yang berisi omset yang mana di dalamnya terdapat hasil dari perkalian antara kolom harga_produk dengan banyaknya_terjual. Penambahan kolom ini dilakukan agar mempermudah dalam menghitung pendapatan minimum dan maksimum.

In [6]:
# Mencari lower dan upper limit pada kolom omset dengan confidence level 95%
std = dataClean['omset'].std()
N = len(dataClean)
low, up = stats.norm.interval(0.95,loc=dataClean['omset'].mean(),scale=std/np.sqrt(N))
print('Lower Limit:',low)
print('Upper Limit:',up)

Lower Limit: 6112221.922601655
Upper Limit: 10515349.210346323


Dari data di atas kita bisa mendapat kesimpulan bahwa potensi maksimum pendapatan yang akan kita dapat dari menjual seblak adalah Rp.10.515.349. dan potensi minimum pendapatannya adalah Rp.6.112.221.

### Perbandingan harga seblak di Jabodetabek dan Luar Jabodetabek.

Diperlukan untuk mencari apakah terdapat perbedaan harga seblak di jabodetabek dan luar jabodetabek dikarenakan harga bahan baku yang berbeda. Untuk menganalisa perbedaan tersebut, digunakan hipotesis sebagai berikut:

<b>H0 = Harga seblak di jabodetabek sama dengan harga di luar jabodetabek </b>

<b>H1 = Harga seblak di jabodetabek tidak sama dengan harga di luar jabodetabek </b>

Uji hipotesis digunakan untuk mengumpulkan bukti dari data yang dapat mendukung atau menolak hipotesis nol. Jika hasil uji menunjukkan ada perbedaan signifikan antara harga seblak di jabodetabek dan luar jabodetabek, kita dapat menolak hipotesis nol dan menerima hipotesis alternatif. Sebaliknya, jika tidak ada bukti yang cukup, kita tidak dapat menolak hipotesis nol.

In [7]:
# Membuat sebuah variabel untuk harga produk di dalam jabodetabek dan produk di luar jabodetabek
dalam_jabodetabek = dataClean[dataClean['jabodetabek'] == 'Jabodetabek']['harga_produk']
luar_jabodetabek= dataClean[dataClean['jabodetabek'] == 'Luar Jabodetabek']['harga_produk']


# Menghitung rata-rata harga seblak di jabodetabek dan luar jabodetabek
print(f"Rata-rata harga seblak di jabodetabek = Rp.{dalam_jabodetabek.mean()}")
print(f"Rata-rata harga seblak di luar jabodetabek adalah = Rp.{luar_jabodetabek.mean()}")

Rata-rata harga seblak di jabodetabek = Rp.17333.167857142857
Rata-rata harga seblak di luar jabodetabek adalah = Rp.21148.272727272728


Membuat sebuah variabel untuk kota yang didalam jabodetabek dan diluar jabodetabek dan melakukan perhitungan rata-rata harga produk. Meskpun terlihat perbedaan, hal ini belum bisa membuktikan secara pasti adanya perbedaan harga. Untuk itu two sample independent test digunakan untuk membandingkan rata-rata dari dua keloimpok data yang berbeda secara independen.

In [8]:
# Melakukan two independent test
t_stat, p_val = stats.ttest_ind(dalam_jabodetabek, luar_jabodetabek)
print('T-Statistic:', t_stat)
print('P-value:', p_val)

T-Statistic: -1.9670344260467834
P-value: 0.04998258288671841


Nilai T-Statistic adalah -1.9670344260467834 yang merupakan negatif bisa mengindikasikan bahwa harga di jabodetabek lebih rendah dibandingkan luar jabodetabek. Dengan nilai P-value 0.04998258288671841 yang lebih kecil dari tingkat signifikansi 0.05 berarti kita memiliki bukti untuk menolak hipotesis 0. Dapat disimpulkan bahwa harga di jabodetabek tidak sama dengan harga di luar jabodetabek.

### Melakukan korelasi untuk mengetahui apakah harga produk mempengaruhi penjualan produk

Untuk mencari tahu apakah harga produk mempengaruhi banyaknya penjualan produk. Untuk menganalisa hal tersebut, digunakan hipotesis sebagai berikut:

<b>H0 = Tidak terdapat korelasi antara harga produk dan banyaknya produk terjual </b>

<b>H1 = Terdapat korelasi antara harga produk dan banyaknya produk terjual</b>

In [9]:
# Memperhitungkan korelasi
corr_rho, pval_s = stats.spearmanr(dataClean['harga_produk'], dataClean['banyaknya_terjual'])

print(f"rho-correlation: {corr_rho:.2f}, p-value: {pval_s}")

rho-correlation: -0.49, p-value: 4.581838831420978e-22


Korelasi Spearman (rho): -0.49, p-value: 4.581838831420978e-22

Spearman digunakan karena tidak terlalu sensitif terhadap outlier, dimana pada data yang kita analisis memiliki banyak outlier.
Nilai rho-correlation -0.49 menunjukkan adanya korelasi negatif yang kuat antara harga produk dan banyaknya produk yang terjual yang berarti korelasi berbanding terbalik. Artinya, ketika harga produk naik, banyaknya produk yang terjual cenderung turun.
Dengan nilai p-value 4.581838831420978e-22 lebih kecil dari tingkat signifikansi 0.05 berarti cukup bukti untuk menolak hipotesis 0 dan berarti terdapat korelasi antara harga produk dan banyaknya produk terjual.

## Kesimpulan

Dari analisis diatas dapat diambil kesimpulan bahwa:  
1. Setelah melakukan perhitungan rata-rata, median, standar deviasi, skewness, dan kurtosis dari kolom harga, banyak produk terjual, dan rating. Dapat terlihat bahwa seblak memiliki tingkat animo yang tinggi di masyarakat.
2. Setelah menghitung potensi minimum dan maksimum pendapatan penjualan produk seblak menggunakan confidence interval, dapat terlihat bahwa kita bisa mendapatkan banyak pendapatan dari menjual seblak karena harga jual yang lumayan tinggi.
3. Hasil dari hipotesis adalah harga di jabodetabek tidak sama dengan harga di luar jabodetabek.
4. Dapat juga disimpulkan bahwa harga produk yang tinggi membuat banyaknya produk yang terjual cenderung turun.

Dengan mempertimbangkan keempat poin diatas dapat diambil kesimpulan bahwa, seblak memiliki potensi yang besar karena animo masyarakat yang tinggi. Tetapi, kita harus bisa mengatur strategi harga karena harga berpengaruh untuk memaksimalkan keuntungan dan jumlah produk yang bisa dijual. Solusinya adalah menjual produk dengan harga murah untuk meningkatkan pendapatan. Selain itu, perbedaan harga antara jabodetabek dengan luar jabodetabek juga bisa dijadikan pertimbangan ketika kita ingin menjual produk seblak. Dengan kesimpulan ini, saya memutuskan untuk melakukan dropshipping produk seblak untuk menghasilkan pendapatan tambahan.